contracts/Rentals.sol

```diff
@@ -303,33 +303,40 @@ contract Rentals is
         }
     }
 
-    /// @notice Set the operator of a given asset.
+    /// @notice Set the update operator of the provided assets.
     /// @dev Only when the rent is active a tenant can change the operator of an asset.
     /// When the rent is over, the lessor is the one that can change the operator.
     /// In the case of the lessor, this is useful to update the operator without having to claim the asset back once the rent is over.
-    /// @param _contractAddress The contract address of the asset.
-    /// @param _tokenId The token id of the asset.
-    /// @param _operator The address that will have operator privileges over the asset.
+    /// Elements in the param arrays correspond to each other in the same index.
+    /// For example, asset with address _contractAddresses[0] and token id _tokenIds[0] will be set _operators[0] as operator.
+    /// @param _contractAddresses The contract addresses of the assets.
+    /// @param _tokenIds The token ids of the assets.
+    /// @param _operators The addresses that will have operator privileges over the given assets.
     function setUpdateOperator(
-        address _contractAddress,
-        uint256 _tokenId,
-        address _operator
+        address[] memory _contractAddresses,
+        uint256[] memory _tokenIds,
+        address[] memory _operators
     ) external nonReentrant {
-        IERC721Rentable asset = IERC721Rentable(_contractAddress);
+        require(
+            _contractAddresses.length == _tokenIds.length && _contractAddresses.length == _operators.length,
```

Comment:
moving `_contractAddresses.length` to a variable and using that for checks only saves 10 gas